medmodels/treatment_effect/matching/propensity.py

```diff
@@ -78,6 +80,10 @@ def match_controls(
             Set[NodeIndex]:  Node Ids of the matched controls.
         """
         # Preprocess the data
+        if one_hot_covariates is None:
```

Comment:
In general, how will this work in the documentation? I get that lists are bad as an input because they are mutable and it could cause problems when calling the function multiple times, but will we just have the default values mentioned in the docstring like it is here?

Reply:
imo this does not need to be specially documented. It does not matter to the user wether the default value is a `[]` or a `None`. The user just needs to know that he/she can pass a list or nothing and that is achieved in both versions of the typing

Reply:
In case of an empty list, yes. But here for example we have lists full of default values that will be set in case of no input.

Reply:
Maybe we should remove these default values completely anyways, I'm not sure if it makes sense, since Attribute naming is so free. But that is not the topic of this PR.

Reply:
You're right, sorry. Was on my phone and didn't check the whole code.
Then I'll pass this on for @FloLimebit to answer once he's back from his vacation.